File: Final Project/Common/src/main/java/commonproject/model/order/OrderDetail.java
package commonproject.model.order;


import commonproject.enumorator.Status;
import commonproject.model.menu.Menu;
import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.math.BigDecimal;

@Entity
@Data
@Table(name = "orderDetail")
public class OrderDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    @Enumerated(EnumType.STRING)
    private Status status;

    @NotNull
    private int menu;

    @NotNull
    private int orderId;

    @NotNull
    private int quantity;

    @NotNull
    private BigDecimal price;

}

File: Final Project/Restaurant-UI/Restaurant-Management/src/app/orderDetail.ts
export class OrderDetail{

    public status:string;
    public menu:number;
    public orderId:number;
    public quantity:number;
    public price:number;

    constructor( status :string,menu:number,orderId:number,quantity:number,price:number){
        this.status = status;
        this.menu=menu;
        this.orderId=orderId;
        this.quantity=quantity;
        this.price=price;
    }
}